src/main/scala/is/hail/expr/Type.scala

```diff
@@ -555,8 +555,12 @@ final case class TVariable(name: String, var t: Type = null) extends Type {
 
   override def unify(concrete: Type): Boolean = {
     if (t == null) {
-      t = concrete
-      true
+      if (predicate(concrete)) {
```

Comment:
Should TVariable ever match non-realizable?  I don't think so.

Reply:
I suppose not.